src/main/java/net/sf/jabref/logic/importer/fileformat/RisImporter.java

```diff
@@ -43,28 +45,17 @@ public String getDescription() {
 
     @Override
     public boolean isRecognizedFormat(BufferedReader reader) throws IOException {
-        // Our strategy is to look for the "AU  - *" line.
-        String str;
-        while ((str = reader.readLine()) != null) {
-            if (RECOGNIZED_FORMAT_PATTERN.matcher(str).find()) {
-                return true;
-            }
-        }
-        return false;
+        // Our strategy is to look for the "TY  - *" line.
+        return reader.lines().anyMatch(line -> RECOGNIZED_FORMAT_PATTERN.matcher(line).find());
     }
 
     @Override
     public ParserResult importDatabase(BufferedReader reader) throws IOException {
         List<BibEntry> bibitems = new ArrayList<>();
-        StringBuilder sb = new StringBuilder();
 
-        String line;
-        while ((line = reader.readLine()) != null) {
-            sb.append(line);
-            sb.append('\n');
-        }
+        String linesAsString = reader.lines().reduce((line, nextline) -> line + "\n" + nextline).get();
 
-        String[] entries = sb.toString().replace("\u2013", "-").replace("\u2014", "--").replace("\u2015", "--")
+        String[] entries = linesAsString.replace("\u2013", "-").replace("\u2014", "--").replace("\u2015", "--")
```

Comment:
Maybe you could extract that method to the StringUtil class, e.g. call it ReplaceUnicodeDashes..